warn/warn_bazel.go

```diff
@@ -260,6 +260,48 @@ func ctxActionsWarning(f *build.File, fix bool) []*Finding {
 	return findings
 }
 
+func fileTypeWarning(f *build.File, fix bool) []*Finding {
+	findings := []*Finding{}
+	toReplace := map[build.Expr]build.Expr{}
+
+	var walk func(e *build.Expr, env *bzlenv.Environment)
+	walk = func(e *build.Expr, env *bzlenv.Environment) {
+		if call, ok := (*e).(*build.CallExpr); ok {
+			if ident, ok := (call.X).(*build.Ident); ok && ident.Name == "FileType" {
+				if binding := env.Get("FileType"); binding == nil {
+					if fix {
+						if len(call.List) > 0 {
+							// Assuming that the first and the only argument of FileType is a list of strings.
+							arg := call.List[0]
+							if binary, ok := arg.(*build.BinaryExpr); ok && binary.Op == "=" {
+								// Assuming that it's a named argument `types`
+								arg = binary.Y
+							}
+							toReplace[*e] = arg
+							return
+						}
+					}
+					start, end := call.Span()
+					findings = append(findings,
+						makeFinding(f, start, end, "filetype",
+							"The FileType function is deprecated, use lists instead.", true, nil))
```

Comment:
Unfortunately, it's not the only use-case. The other use-case is FileType.filter (https://docs.bazel.build/versions/master/skylark/lib/FileType.html#filter).
Updating the code to use lists is not always working. Maybe we should be more careful and avoid the automated fix?

Reply:
Removed the fix functionality, just a warning now.